src/System.Device.Gpio/System/Device/Gpio/RaspberryBoardInfo.cs

```diff
@@ -99,7 +99,7 @@ private RaspberryBoardInfo(Dictionary<string, string> settings)
             ProcessorName = _settings.TryGetValue("Hardware", out string? hardware) && hardware is object ? hardware : string.Empty;
 
             if (_settings.TryGetValue("Revision", out string? revision)
-                && !string.IsNullOrEmpty(revision)
+                && revision is { Length: > 0 }
```

Comment:
I personally find this less readable

Reply:
We can certainly change it back.

I like the new syntax for three reason:

- It is centered on the object, as opposed to a helper. I do, however, agree that the name of helper very clearly describes the check being performed. This is clearly the reason why people prefer this approach. The mental cost to figure out the nature of the check is 0.
- The syntax is symmetric with `revision is object` (or `revision is != null`). The `{ }` is the same as and takes the place of `object` in the check, with the added benefit that you can perform property tests. I think this becomes more natural when you start using `is` generally instead of `==` and `!=`. Related: https://twitter.com/jaredpar/status/1115019017297596416.
- It is shorter.